lowranceusr.cc

```diff
@@ -406,15 +938,35 @@ lowranceusr_parse_waypt(Waypoint* wpt_tmp)
     wpt_tmp->SetCreationTime(base_time_secs + waypt_time);
   }
 
-  if (global_opts.debug_level >= 2) {
-    printf(MYNAME " parse_waypt: creation time %d\n",
-           (int)wpt_tmp->creation_time.toTime_t());
-    printf(MYNAME " parse_waypt: base_time %d\n", (int)base_time_secs);
-    printf(MYNAME " parse_waypt: waypt time %d\n", (int)waypt_time);
+  if (global_opts.debug_level > 2) {
+    if (global_opts.debug_level == 99) {
+      QDateTime dt;
+      QTime t;
+      dt = wpt_tmp->GetCreationTime();
+      t  = wpt_tmp->GetCreationTime().time();
+      printf(" %s %s", qPrintable(dt.toString("yyyy/MM/dd")), qPrintable(t.toString("hh:mm")));
+    } else {
+      printf(MYNAME " parse_waypt: creation time %d, base_time %d waypt_time %d\n",
+             (int)wpt_tmp->creation_time.toTime_t(),
```

Comment:
Notably qDebug knows how to print a QDateTime in a way that's human-readable. qDebug() << "created" << wpt_tmp->creation_time << "base time" << base_time_secs should do something reasonable.
It's not really suited to the writing of the files, but it saves that ickiness in both of these two printf lines.